Frameworks/UIKit/UINibUnarchiver.mm

```diff
@@ -203,7 +203,13 @@ static id constructObject(UINibUnarchiver* self, Object* pObj) {
         id classId = pObj->classType; // objc_getClass(pObj->className);
         assert(classId != nil);
 
-        pObj->cachedId = [classId alloc];
+        // TODO: #1105 alloc without init is invalid, and not all classes support alloc-init anyway
+        // look for a proper fix to this issue, rather than just special-casing for UIFont
+        if ([classId isEqual:[UIFont class]]) {
+            pObj->cachedId = [[UIFont defaultFont] retain];
```

Comment:
Hm. After a little investigation, it looks like our Xib2Nib and UIFont implementation are a little out of spec. NSFont does indeed implement NSCoding, but it does seem like UIFont doesn't, and so my only conclusion is we're only ever supposed to use UIFontDescriptor to encode/decode on iOS. IE, in `UILabel -initWithCoder`, `_font` should be populated with `[UIFont fontWithDescriptor:[_coder decodeObjectForKey:@"UIFont[Descriptor]" pointSize:0.0f]` and of course we'd have to update Xib2Nib to output the right thing. 

Since we're no longer investing in Xib2Nib, I'd say for this is a good enough solution, so long as the new UIFont does still implements initWithCoder.